src/utils.js

```diff
@@ -56,41 +56,189 @@ export function getEntriesFromXHR(file) {
         );
       })
   );
-
   return [request, promise];
 }
 
 export async function blobToDataUrl(blob) {
   return new Promise((resolve, reject) => {
     const reader = new FileReader();
-
     reader.onload = event => {
       resolve(event.target.result);
     };
-
     reader.readAsDataURL(blob);
   });
 }
 
 export function getResourceHref(resource) {
   const resourceHrefAttribute = resource.getAttribute("href");
-
   if (resourceHrefAttribute) {
     return resourceHrefAttribute;
   }
-
   const fileNode = resource.querySelector("file");
-
   const fileNodeHrefAttribute =
     fileNode != null && fileNode.getAttribute("href");
-
   if (fileNodeHrefAttribute) {
     return fileNodeHrefAttribute;
   }
-
   return null;
 }
 
 export function getExtension(uri) {
   return uri.split(".").pop();
 }
+
+function $text(document, selector) {
+  const node = document.querySelector(
+    "metadata > lom > general > title > string"
+  );
+  return node && node.textContent;
+}
+
+export function getResourcesFromXml(xml) {
+  const parser = new DOMParser();
+  const manifest = parser.parseFromString(xml, "text/xml");
+  const title = $text(manifest, "metadata > lom > general > title > string");
+  const schema = $text(manifest, "metadata > schema");
+  const schemaVersion = $text(manifest, "metadata > schemaversion");
+  const rightsDescription = $text(
+    manifest,
+    "metadata > lom > rights > description"
+  );
+  const resources = Array.from(
```

Comment:
it seems like mutation may be a better option to remove the needless re-iteration of the same Array.